GitUI/BranchTreePanel/RepoObjectsTree.Nodes.cs

```diff
@@ -289,11 +369,22 @@ protected async Task ReloadNodesAsync(Func<CancellationToken, Func<RefsFilter, I
 
                 await treeView.SwitchToMainThreadAsync(token);
 
+                // remember multi-selected nodes
+                var multiSelected = this.GetMultiSelection().Select(node => node.GetHashCode()).ToArray();
+
                 Nodes.Clear();
                 Nodes.AddNodes(newNodes);
 
+                // re-apply multi-selection
+                if (multiSelected.Length > 0)
+                {
+                    this.GetNodesAndSelf().Where(node => multiSelected.Contains(node.GetHashCode()))
+                        .ForEach(node => node.IsSelected = true);
+                }
```

Comment:
This is rather inefficient (i.e., O(n^2)) as we reiterate `multiSelected` over and over in a loop. IIUIC `multiSelected` is a lookup array of int, so it should be made a `HashSet<int>` which is O(1) for read.
Also a foreach loop has a better perf than LINQ, and I recommend rewriting it as a foreach loop.